lib/puppet/provider/package/pkgng.rb

```diff
@@ -0,0 +1,132 @@
+require 'puppet/provider/package'
+
+Puppet::Type.type(:package).provide :pkgng, :parent => Puppet::Provider::Package do
+  desc "A PkgNG provider for FreeBSD and DragonFly."
+
+  commands :pkg => "/usr/local/sbin/pkg"
+
+  confine :operatingsystem => [:freebsd, :dragonfly]
+  confine :pkgng_enabled => :true
+
+  defaultfor :operatingsystem => :freebsd
```

Comment:
This should be the default for DragonFlyBSD as well, pkgsrc hasn't been used in some time and we use pkgng exclusively.